5_databases/week3/hw-w3.sql

```diff
@@ -0,0 +1,182 @@
+CREATE DATABASE mealsharing;
+
+Use mealsharing;
+
+CREATE TABLE
+    `reservation` (
+        `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+        `number_of_guests` INT NOT NULL,
+        `meal_id` int(10) UNSIGNED,
+        `created_date` DATE NOT NULL,
+        `contact_phonenumber` VARCHAR(255) NOT NULL,
+        `contact_name` VARCHAR(255) NOT NULL,
+        `contact_email` VARCHAR(255) NULL,
+        CONSTRAINT `fk_meal` FOREIGN KEY (`meal_id`) REFERENCES `meal` (`id`) ON DELETE CASCADE
+    ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+
+CREATE TABLE
+    `meal` (
+        `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+        `title` VARCHAR(255) NOT NULL,
+        `description` TEXT,
+        `location` VARCHAR(255) NOT NULL,
+        `when` DATETIME NOT NULL,
+        `max_reservations` int NOT NULL,
+        `price` DECIMAL NOT NULL,
+        `created_date` DATE NOT NULL
+    ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+
+CREATE TABLE
+    `review` (
+        `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+        `title` VARCHAR(255) NOT NULL,
+        `description` TEXT,
+        `meal_id` int(10) UNSIGNED,
+        `stars` int NOT NULL,
+        `created_date` DATE,
+        CONSTRAINT `fk_meal_review` FOREIGN KEY (`meal_id`) REFERENCES `meal` (`id`) ON DELETE CASCADE
+    ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+
+-- Meals
+
+INSERT INTO
+    meal (
+        title,
+        description,
+        location,
+        `when`,
+        max_reservations,
+        price,
+        created_date
+    )
+VALUES (
+        'Chicken curry',
+        'tender and yummy',
+        'Copenhagen',
+        '2024-10-25 06:54:16',
+        4,
+        200,
+        '2023-09-10 11:05:05'
+    ), (
+        'Lamb curry',
+        'Cooked with love',
+        'London',
+        '2024-01-21 06:54:16',
+        9,
+        150,
+        '2023-10-15 15:05:09'
+    ), (
+        'Roast lamb',
+        'Roasted with even more love',
+        'Paris',
+        '2024-01-15 19:23:09',
+        6,
+        800,
+        '2022-12-10 02:02:01'
+    ), (
+        'Smoked salmon',
+        'Imported from Norway',
+        'Copenhagen',
+        '2024-08-10 06:54:16',
+        10,
+        300,
+        '2023-02-11 23:10:03'
+    ), (
+        'Enchilada',
+        'You probably wouldn''t like that',
+        'Copenhagen',
+        '2024-04-03 05:34:11',
+        6,
+        900,
+        '2022-10-15 10:09:09'
+    );
```

Comment:
Good that you don't set the id explicitly. 

Normally you would see people put all values for a single row on one line to make it a bit more compact. It has no functional change.
```sql 
INSERT INTO meal (title, description, location, `when`, max_reservations, price, created_date)
VALUES 
   ('Chicken curry', 'tender and yummy', 'Copenhagen', '2024-10-25 06:54:16', 4, 200, '2023-09-10 11:05:05'), 
   ('Lamb curry', 'Cooked with love', 'London', '2024-01-21 06:54:16', 9, 150, '2023-10-15 15:05:09'),
    ...
```